File: src/test/java/me/pcasaes/hexoids/service/kafka/converter/UUIDBytesSerializerDeserializerTest.java
package me.pcasaes.hexoids.service.kafka.converter;

import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNotSame;

class UUIDBytesSerializerDeserializerTest {

    @Test
    void tessUUIDSerializerAndDeserializer() {
        UUID one = UUID.randomUUID();

        byte[] oneBytes = new UUIDBytesSerializer().serialize("", one);
        assertNotNull(oneBytes);
        assertEquals(Long.BYTES * 2, oneBytes.length);

        UUID oneDeserialized = new UUIDBytesDeserializer().deserialize("", oneBytes);
        assertNotNull(oneDeserialized);

        assertEquals(one, oneDeserialized);
        assertNotSame(one, oneDeserialized);
    }
}

File: src/test/kotlin/me/pcasaes/hexoids/infrastructure/kafka/converter/UUIDBytesSerializerDeserializerTest.kt
package me.pcasaes.hexoids.infrastructure.kafka.converter

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.lang.Long
import java.util.UUID

class UUIDBytesSerializerDeserializerTest {
    @Test
    fun tessUUIDSerializerAndDeserializer() {
        val one = UUID.randomUUID()

        val oneBytes = UUIDBytesSerializer().serialize("", one)
        Assertions.assertNotNull(oneBytes)
        Assertions.assertEquals(Long.BYTES * 2, oneBytes.size)

        val oneDeserialized = UUIDBytesDeserializer().deserialize("", oneBytes)
        Assertions.assertNotNull(oneDeserialized)

        Assertions.assertEquals(one, oneDeserialized)
        Assertions.assertNotSame(one, oneDeserialized)
    }
}